passageflex/src/main/java/id/passage/passageflex/RegisterException.kt

```diff
@@ -0,0 +1,119 @@
+import id.passage.android.passageflex.model.Model401Code
+import id.passage.passageflex.client.infrastructure.ClientException
+import id.passage.passageflex.client.infrastructure.ServerException
+
+/**
+ * Thrown when there's an error registering a user with a passkey.
+ *
+ * @see RegisterCancellationException
+ * @see RegisterInterruptedException
+ * @see RegisterConfigurationException
+ * @see RegisterNoCreateOptionException
+ * @see RegisterUnsupportedException
+ * @see RegisterInvalidRequestException
+ * @see RegisterCredentialException
+ * @see RegisterServerException
+ */
+public open class RegisterException(message: String) : PassageFlexException(message) {
+    internal companion object {
+        internal fun convert(e: Exception): RegisterException {
+            val message = e.message ?: e.toString()
+            return when (e) {
+                is CreateCredentialException -> convertCreateCredentialException(e)
+                is ClientException -> convertClientException(e)
+                is ServerException -> RegisterServerException(message)
+                else -> RegisterException(message)
+            }
+        }
+
+        private fun convertClientException(e: ClientException): RegisterException {
+            val error = parseClientException(e)
+            val message = error?.error ?: ""
+            return when (error?.code) {
+                Model400Code.request.toString() -> {
+                    RegisterInvalidRequestException(message)
+                }
+                Model401Code.webauthnLoginFailed.toString() -> {
+                    RegisterCredentialException(message)
+                }
+                else -> {
+                    RegisterException(message)
+                }
+            }
+        }
+
+        private fun convertCreateCredentialException(e: CreateCredentialException): RegisterException {
+            val message = e.message ?: e.toString()
+            return when (e) {
+                is CreateCredentialCancellationException -> {
+                    RegisterCancellationException(message)
+                }
+                is CreateCredentialInterruptedException -> {
+                    RegisterInterruptedException(message)
+                }
+                is CreateCredentialProviderConfigurationException -> {
+                    RegisterConfigurationException(message)
+                }
+                is CreateCredentialNoCreateOptionException -> {
+                    RegisterNoCreateOptionException(message)
+                }
+                is CreateCredentialUnsupportedException -> {
+                    RegisterUnsupportedException(message)
+                }
+                else -> RegisterException(message)
+            }
+        }
+    }
+}
+
+/**
+ * The user intentionally canceled the operation and chose not to register the credential.
+ */
+public class RegisterCancellationException(message: String) : RegisterException(message)
```

Comment:
How do you determine this on mobile? I'd love to handle this on the JS side but not sure how atm

Reply:
The Android Credential Manager explicitly throws a cancelation exception that I can then transform into this custom exception :)

Reply:
Interesting. I wonder if I could intercept that on the web. The element for example just shows our standard failure UI when that happens but that would be up to the developer to build for JS. Will give it a thought!